Plugin/Controller/Adminhtml/Page/MassDeletePlugin.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mooore\WordpressIntegrationCms\Plugin\Controller\Adminhtml\Page;
+
+use Magento\Cms\Controller\Adminhtml\Page\MassDelete as MassDeleteController;
+use Magento\Ui\Component\MassAction\Filter;
+use Magento\Cms\Model\ResourceModel\Page\CollectionFactory;
+use Mooore\WordpressIntegrationCms\Model\RemotePageRepository;
+
+class MassDeletePlugin
+{
+    /**
+     * @var Filter
+     */
+    private $filter;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * @var RemotePageRepository
+     */
+    private $remotePageRepository;
+
+    public function __construct(
+        Filter $filter,
+        CollectionFactory $collectionFactory,
+        RemotePageRepository $remotePageRepository
+    ) {
+        $this->filter = $filter;
+        $this->collectionFactory = $collectionFactory;
+        $this->remotePageRepository = $remotePageRepository;
+    }
+
+    /**
+     * This function calls to Wordpress to delete the Magento URL entry in a MassDelete action.
+     *
+     * @param MassDeleteController $subject
+     * @return void
+     */
+    public function beforeExecute(MassDeleteController $subject)
```

Comment:
Saving the metadata should only happen when the module has been configured to do so! Also need to check if the API credentials are not empty.

Reply:
Same answer as in the DeletePlugin: The checking of the API credentials happens further down the line in the RemotePageRepository. An error will occur and will be written to the log files for further investigation. Could you explain to me what configuration you would expect to see here?